app/src/org/commcare/android/logic/BarcodeScanListenerDefaultImpl.java

```diff
@@ -31,7 +31,7 @@ public static void onBarcodeResult(BarcodeScanListener barcodeScanListener, int
         //endregion
 
         if (resultCode == Activity.RESULT_OK) {
-            String result = intent.getStringExtra(SCAN_RESULT);
```

Comment:
Hmm... I _think_ we do because the barcode scan has returned status "RESULT_OK" which would be odd to do for a null response when you're explicity expecting a result. I suppose this is more contractual than hard-code though... maybe I'll add some null checking for good measure